chain/client/src/sync/state.rs

```diff
@@ -800,6 +727,245 @@ impl StateSync {
             _ => {}
         }
     }
+
+    /// Checks if the header is downloaded.
+    /// If the download is complete, then moves forward to StateDownloadParts, otherwise retries the header request.
+    /// Returns (download_timeout, need_shard).
```

Comment:
nit: instead of giving only the variable names, it could be extra nice to describe their semantics. I guess the names are almost self-explanatory. But not so much for someone (like me) unfamiliar with the state sync code.

Reply:
> self-explanatory

No they were not :laughing: 
Re-read the code, gave more descriptive names to the variables, added a comment explained their semantics.